cmd/controller/main.go

```diff
@@ -102,15 +99,6 @@ func main() {
 		logger.Fatalf("Error building shared clientset: %v", err)
 	}
 
-	// TODO: Rip this out from all the controllers since we can get it
-	// from provider.
-	// Build a rest.Config from configuration injected into the Pod by
-	// Kubernetes. Clients will use the Pod's ServiceAccount principal.
-	restConfig, err := rest.InClusterConfig()
```

Comment:
I don't see where this gets moved to get it from the provider. Does this just happen automatically?

Reply:
I can remove the `cfg` parameter from the constructor and have each constructor:

```go
cfg, err := controllerruntime.GetConfig()
if err != nil {
	logger.Fatalf("Error building kubeconfig: %v", err)
}
```

My preference is to do it once here in main and pass it in. It is slightly less code for each constructor and ensures they all get identical configs (at the cost of an extra parameter).

Thoughts?

Reply:
✌️ move it into ⚰️